packages/experimental/RadioGroup/src/Radio/useRadio.ts

```diff
@@ -0,0 +1,86 @@
+import { RadioProps, RadioState } from './Radio.types';
+import * as React from 'react';
+import { useRadioGroupContext } from '../RadioGroup/radioGroupContext';
+import { useAsPressable, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { radioSelectActionLabel } from './Radio.styling';
+
+export const useRadio = (props: RadioProps): RadioState => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    value,
+    disabled,
+    accessibilityLabel,
+    componentRef = defaultComponentRef,
+    accessibilityPositionInSet,
+    accessibilitySetSize,
+    ...rest
+  } = props;
+
+  // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+  const info = useRadioGroupContext();
+
+  const buttonRef = useViewCommandFocus(componentRef);
+
+  /* We don't want to call the user's onChange multiple times on the same selection. */
+  const changeSelection = React.useCallback(() => {
+    if (value != info.value) {
+      info.onChange && info.onChange(value);
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
+
+  /* We use the componentRef of the currently selected button to maintain the default tabbable
+    element in a RadioGroup. Since the componentRef isn't generated until after initial render,
+    we must update it once here. */
+  React.useEffect(() => {
+    if (value == info.value) {
+      info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+    }
+  }, [info, value, componentRef]);
+
+  // Ensure focus is placed on button after click
+  const changeSelectionWithFocus = useOnPressWithFocus(componentRef, changeSelection);
+
+  /* RadioButton changes selection when focus is moved between each RadioButton and on a click */
+  const pressable = useAsPressable({
+    ...rest,
+    onPress: changeSelectionWithFocus,
+    onFocus: changeSelection,
```

Comment:
Does FocusZone make sure that if you press "tab" while in the focus zone, that you leave the focus zone?

Reply:
Yes it does.